Classes/DataProcessing/LanguageMenuProcessor.php

```diff
@@ -0,0 +1,494 @@
+        $this->menuConfig['1.']['ACT'] = '1';
+        $this->menuConfig['1.']['ACT.'] = $this->menuConfig['1.']['NO.'];
+        $this->menuConfig['1.']['ACT.']['stdWrap.']['cObject.']['30.']['value'] = '1';
+        $this->menuConfig['1.']['CUR'] = '1';
+        $this->menuConfig['1.']['CUR.'] = $this->menuConfig['1.']['ACT.'];
+        $this->menuConfig['1.']['CUR.']['stdWrap.']['cObject.']['40.']['value'] = '1';
+        $this->menuConfig['1.']['USERDEF1'] = $this->menuConfig['1.']['NO'];
+        $this->menuConfig['1.']['USERDEF1.'] = $this->menuConfig['1.']['NO.'];
+        $this->menuConfig['1.']['USERDEF1.']['stdWrap.']['cObject.']['50.']['value'] = '0';
+        $this->menuConfig['1.']['USERDEF2'] = $this->menuConfig['1.']['ACT'];
+        $this->menuConfig['1.']['USERDEF2.'] = $this->menuConfig['1.']['ACT.'];
+        $this->menuConfig['1.']['USERDEF2.']['stdWrap.']['cObject.']['50.']['value'] = '0';
+    }
+
+    /**
+     * @param ContentObjectRenderer $cObj The data of the content element or page
+     * @param array $contentObjectConfiguration The configuration of Content Object
+     * @param array $processorConfiguration The configuration of this processor
+     * @param array $processedData Key/value store of processed data (e.g. to be passed to a Fluid View)
+     * @return array the processed data as key/value store
+     */
+    public function process(ContentObjectRenderer $cObj, array $contentObjectConfiguration, array $processorConfiguration, array $processedData)
+    {
+        $this->cObj = $cObj;
+        $this->processorConfiguration = $processorConfiguration;
+
+        // Get Configuration
+        $this->menuTargetVariableName = $this->getConfigurationValue('as');
+        $this->menuAlternativeSortingField = $this->getConfigurationValue('alternativeSortingField');
+
+        // Validate Configuration
+        $this->validateConfiguration();
+
+        // Build Configuration
+        $this->prepareConfiguration();
+        $this->buildConfiguration();
+
+        // Process Configuration
+        $menuContentObject = $cObj->getContentObject('HMENU');
+        $renderedMenu = $menuContentObject->render($this->menuConfig);
+        if (!$renderedMenu) {
+            return $processedData;
+        }
+
+        // Process menu
+        $menu = json_decode($renderedMenu, true);
+        $processedMenu = [];
+
+        foreach ($menu as $key => $page) {
+            $processedMenu[$key] = $this->processAdditionalDataProcessors($page, $processorConfiguration);
+        }
+
+        // Return processed data
+        $processedData[$this->menuTargetVariableName] = $processedMenu;
+        return $processedData;
+    }
+
+    /**
+     * Process additional data processors
+     *
+     * @param array $page
+     * @param array $processorConfiguration
+     */
+    protected function processAdditionalDataProcessors($page, $processorConfiguration)
+    {
+        if (is_array($page['children'])) {
```

Comment:
Children will not be available here, can be removed.

Reply:
as far as i see the processor makes currently no sense. once HMENU or i more prefer a new content object in the core because we are creating flat menus here and nothing hierarchical has changed maybe we can use it to load e.g. flags.

Reply:
flags could now also be loaded easily via css, we can prepare selectors for that in the markup - i would drop as mentioned before additional data processors.